### PR TITLE
chore(ui): add support for service documentation (part-1)

### LINKED ISSUE #6981: All ingestion connector form should display relevant docs
**Feature**
Our ingestion connectors should display relevant long form docs.
example https://openmetadata.slack.com/archives/C02B6955S4S/p1661559688634869
@shahsank3t fyi we did this before.
cc @ShilpaVernekar @pmbrull for content

**Describe the task**

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Closes #6981 
### Describe your changes :

This PR adds support for the following things.
- While adding a service show the requirements for the selected service as 2 step
- Support for showing the focused field document on the right-side panel.(Currently only supported for add and edit service)

**Pending**
- Markdown rendering style improvements
  - [x] inconsistent line jumps from the md file
  - [x] links in the right hand side docs are not clickable. UI loses focus of the selected field
  - [x] Fields with internal properties do not render the documentation
- Showing documentation for ingestion workflows ( will be covered in second part of this PR )

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :
image

> Note: The requirements here are a dummy structure for testing purposes.

image

image

> Note: Currently we have the right panel as a sticky element, I will try to make it float next to the focused field.

image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by harshach** [comment]
@Sachin-chaurasiya do we want to keep the text below the labels if there is more documentation
image
@pmbrull can we get some documentation on one of the connectors so that we can see it in action.
What we have is very minimal doesn't represent the actual text that we end up showing

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya do we want to keep the text below the labels if there is more documentation
> 
> image
> 
> @pmbrull can we get some documentation on one of the connectors so that we can see it in action.
> 
> What we have is very minimal doesn't represent the actual text that we end up showing

Yes, we can get rid of that as we are showing the documentation on right side panel.

**Comment by Sachin-chaurasiya** [comment]
@open-metadata/ingestion 

### Adding markdown files for service documentation.

- All the markdown files will be in `openmetadata-ui/src/main/resources/ui/src/public/locales` directory
- locales directory will have subdirectories for different languages, for example, `en-US`, `fr-FR`.
- In the language directory we will have a structure like this

image

Where serviceType should be  `Dashboard , Database, Messaging,  Metadata, MlModel, ObjectStore ,Pipeline, etc` and service name should be the name of the service like `Mysql, Postgres, etc`.

> NOTE: here casing will matter so make sure the file name is `Mysql.md` not `mysql.md` and folder name should be `Database` not `database`.

- To add images, create a `doc-images` folder inside the `locales` directory and put the images there.
> NOTE: The image path should be relative

Example : `image-testing`

cc : @pmbrull

**Review by chirag-madlani — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/AddService/Steps/ServiceRequirements.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      const isEnglishLanguage = i18n.language === SupportedLocales.English;
+      const filePath = `${i18n.language}/${serviceType}/${serviceName}/requirements.md`;
+      const fallbackFilePath = `${SupportedLocales.English}/${serviceType}/${serviceName}/requirements.md`;
+
+      const [translation, fallbackTranslation] = await Promise.allSettled([
+        fetchMarkdownFile(filePath),
+        isEnglishLanguage
+          ? Promise.reject('')
+          : fetchMarkdownFile(fallbackFilePath),
+      ]);
```
[the remark this answers is not in the document]
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/AddService/Steps/ServiceRequirements.tsx:53]: Make sense, I will improve this piece, thanks @chirag-madlani.

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**